strings/slices/slices.go

```diff
@@ -0,0 +1,52 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package slices defines various functions useful with slices of string type.
+// The goal is to be as close as possible to
+// https://github.com/golang/go/issues/45955. Ideal would be if just
+// the import can be updated if the "slices" package becomes standard.
+package slices
+
+// func Filter[T any](d, s []T, keep func(T) bool) []T
+
+// Filter returns a new slice with the strings from the passed slice that are valid.
+
+// Filter appends to d each element e of s for which keep(e) returns true.
+// It returns the modified d. d may be s[:0], in which case the kept
```

Comment:
hmm, I think that this is returning a pointer to the original `d` and mutating it,

Reply:
The comment is taken from https://github.com/golang/go/issues/45955. Looked weird to me but it works as the "inplace" unit-test shows. But `d` must be returned since `s` is mutated but keeps it's original length.

Reply:
my question is if `d` should be returned, hence modifying also `d`, or if a new slice should be returned

Reply:
If that's the semantic Go is choosing, we should stick to it.  It makes sense to me - if you want a copy, pass a copy.  Don't pessimize by default.